controller/scheduler/job.go

```diff
@@ -87,6 +87,10 @@ type Job struct {
 	// referenced from within the main scheduler loop
 	state JobState
 
+	// reason is a human readable indication of why a job is currently
+	// pending. It is only valid while the job is in the pending state.
+	reason string
```

Comment:
Yeah lets go with that. I will refactor this to `statusReason` and look at adding the placement errors etc.